core/domain/exp_services.py

```diff
@@ -1998,28 +2131,93 @@ def update_exploration(
-        user_services.add_edited_exploration_id(committer_id, exploration_id)
-        user_services.record_user_edited_an_exploration(committer_id)
-        if not rights_manager.is_exploration_private(exploration_id):
-            user_services.update_first_contribution_msec_if_not_set(
-                committer_id, utils.get_current_time_in_millisecs())
+        models_to_put.extend(
+            user_services.add_edited_exploration_id(
+                committer_id,
+                exploration_id
+            )
+        )
+        user_settings = user_services.get_user_settings(
+            committer_id,
+            strict=False
+        )
+        if user_settings is not None:
+            updated_user_settings_with_last_edit_info = (
+                user_services.record_user_edited_an_exploration(
+                    user_settings
+                )
+            )
+            if not rights_manager.is_exploration_private(exploration_id) and (
+                user_settings.first_contribution_msec is None
+            ):
+                updated_user_settings_with_first_contribution_msec = (
+                    user_services
+                    .update_first_contribution_msec(
+                        updated_user_settings_with_last_edit_info,
+                        utils.get_current_time_in_millisecs()
+                    )
+                )
+                user_settings_model = (
+                    user_services.convert_to_user_settings_model(
+                        updated_user_settings_with_first_contribution_msec
+                    )
+                )
+                models_to_put.append(user_settings_model)
+            else:
+                models_to_put.append(
+                    user_services.convert_to_user_settings_model(
+                        updated_user_settings_with_last_edit_info
+                    )
+                )
 
     if opportunity_services.is_exploration_available_for_contribution(
             exploration_id):
-        opportunity_services.update_opportunity_with_updated_exploration(
-            exploration_id)
+        models_to_put.extend(
+            opportunity_services
+            .compute_opportunity_models_with_updated_exploration(
+                exploration_id
+            )
+        )
+    exp_rights = exp_models.ExplorationRightsModel.get_by_id(
+        exploration_id
+    )
+    exp_summary_model = exp_models.ExpSummaryModel.get(exploration_id)
+    exp_summary = update_exploration_summary(
+        updated_exploration,
+        exp_rights,
+        exp_fetchers.get_exploration_summary_from_model(exp_summary_model),
+        skip_exploration_model_last_updated=True
+    )
+    exp_summary.version += 1
+    updated_exp_summary_model: exp_models.ExpSummaryModel = (
+        populate_exp_summary_model_fields(
+            exp_summary_model, exp_summary
+        )
+    )
+    models_to_put.append(updated_exp_summary_model)
```

Comment:
> Also, when testing in a production server make sure to do some manual runs saving private/public explorations to see whether anything breaks with the put_multi() in normal request/response operation.

Can you link to the job doc here or in the PR description? It's a prod change so I think that should be reviewed as part of the PR too (will talk to @vojtechjelinek about making this standard practice).